pwiz_tools/Skyline/FileUI/ExportMethodDlg.cs

```diff
@@ -256,6 +261,13 @@ public ExportMethodDlg(SrmDocument document, ExportFileType fileType)
             UpdateInstrumentControls(MethodType);
         }
 
+        private void SelectInstrumentTypeIfValid(string instrumentType, string[] listTypes)
+        {
+            string selectionType = GetSelectedInstrument(instrumentType);
+            if (listTypes.Contains(selectionType) && GetInstrumentTypeFromSelection(selectionType, true) != null)
+                InstrumentType = listTypes.FirstOrDefault(typeName => typeName.Equals(instrumentType));

```

Comment:
This code is a little confusing, because there's "instrumentType", "selectionType" and whatever "GetInstrumentTypeFromSelection(selectionType)" returns.

This particular line is equivalent to:
`InstrumentType = listTypes.Contains(instrumentType) ? instrumentType : null;`

Is that what you intended? (it probably is)
Just wanted to make sure this code is doing what you expect it to.

Reply:
Okay. Good point. I could barely understand it myself, and I am not sure it worked correctly. I have added a bunch of comments, and several tests of the initial selection of the Instrument type combo box, and also improved the code to work exactly the way I was expecting it to, after debug stepping through it during the tests.